informal_update/utils.py

```diff
@@ -0,0 +1,85 @@
+from io import BytesIO
+from django.conf import settings
+from django.template.loader import render_to_string
+from django.contrib.auth.models import User
+
+from xhtml2pdf import pisa
+
+from notifications.notification import send_notification
+from main.frontend import get_project_url
+from .models import InformalUpdate, Donors
+
+
+def render_to_pdf(template_src, context_dict={}):
+    html = render_to_string(template_src, context_dict)
+    result = BytesIO()
+    pdf = pisa.pisaDocument(BytesIO(html.encode("ISO-8859-1")), result)
+    if not pdf.err:
+        file = {
+            'filename': 'informal_update.pdf',
+            'file': result.getvalue()
+        }
+        return file
+    return None
+
+
+def generate_file_data(data):
+    return [
+        {
+            'image': item.get('file') and (settings.BASE_URL + item['file']),
+            'caption': item['caption'],
+        }
+        for item in data
+    ]
+
+
+def send_email_when_informal_update_created(instance):
+    from informal_update.serializers import InformalUpdateSerializer
+
+    informal_update = InformalUpdateSerializer(instance)
+    data = informal_update.data
+    map_list = generate_file_data(data['map_files'])
+    graphics_list = generate_file_data(data['graphics_files'])
+    email_context = {
+        'resource_url': get_project_url(instance.id),
+        'title': data['title'],
+        'situational_overview': data['situational_overview'],
+        'map_list': map_list,
+        'graphic_list': graphics_list,
+        'actions_taken': data['actions_taken'],
+        'resources': data['references'],
+    }
+    print(email_context['resources'])
+    share_with_group = instance.share_with
+    if instance.share_with is None:
+        return
```

Comment:
Let's move this to the top as we don't need to do anything if this is not available.